lib.rs

```diff
@@ -221,12 +221,22 @@ impl From<LayoutErr> for CollectionAllocErr {
         CollectionAllocErr::CapacityOverflow
     }
 }
+impl CollectionAllocErr {
+    #[cold]
+    #[inline(never)]
+    fn bail(&self) -> ! {
```

Comment:
`#[cold]` here looks uncontroversial to me, though maybe the `#[inline(never)]` prevents stuff from getting potentially optimized away?